pkg/db/client.go

```diff
@@ -114,6 +123,14 @@ func (t *TridentDB) InsertCampaign(campaign *Campaign) error {
 	return t.db.Create(campaign).Error
 }
 
+// UpdateCampaign is a required function by the Datastore interface. it is a
+// thin wrapper around the Gorm save method, this is largely to help with
+// database mocking for tests (and for help with multiple drivers in the
+// future).
+func (t *TridentDB) UpdateCampaign(campaign *Campaign) error {
+	return t.db.Save(campaign).Error
+}
+
```

Comment:
Per a comment in `pkg/server/handlers.go`, we could change this interface to take a query and map[string]interface{} for updates.